src/net/socket.rs

```diff
@@ -79,7 +79,9 @@ impl Socket {
 
     pub fn new(domain: Domain, ty: Type, protocol: Option<Protocol>) -> io::Result<Self> {
         let socket = Socket2::new(domain, ty, protocol)?;
-        #[cfg(unix)]
+        // on Linux and Windows we use native completion-based IO interfaces and are not
+        // interested to receive EAGAIN/EWOULDBLOCK as accept result
+        #[cfg(all(unix, not(target_os = "linux")))]
```

Comment:
@George-Miao What's your opinion?

Reply:
Kernel 5.7 is old. Even WSL is 5.15 now. We can use this flag.

Reply:
Yes, it will decrease used resources. Should socket be nonblocking for this feature to work?

I guess blocking flag is relevant only for application and not for kernel thread.

Reply:
> Yes, it will decrease used resources. Should socket be nonblocking for this feature to work?
> 
> I guess blocking flag is relevant only for application and not for kernel thread.

Yes I think this is required

Reply:
The problem is fixed in newer kernels - https://lore.kernel.org/all/a545c1ae-02a7-e7f1-5199-5cd67a52bb1e@kernel.dk/T/#mc720f37164873a8853bb1953515329c242a89a3b

I'm working on 5.15 and tests fail with nonblocking sockets.